release_dashboard/lib/logic/git.dart

```diff
@@ -0,0 +1,121 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:conductor_core/conductor_core.dart';
+
+abstract class GitValidation {
+  /// Returns the regex needed to validate this type of input.
+  RegExp get regex;
+
+  /// Returns the error message when the regex validation fails.
+  String get errorMsg;
+
+  /// Uses [regex] to validate the input provided. Returns true if it is valid, false otherwise.
+  ///
+  /// This method calls [sanitize] before the validation.
+  bool isValidate(String? input);
+
+  /// Removes irrelevant characters such as whitespaces.
+  String sanitize(String? input);
```

Comment:
If this is the same across all functions, why not just implement it here and inherit it in the children?